python/subnet-s3-vpc-endpoint.py

```diff
@@ -0,0 +1,102 @@
+ This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+# 
+# Description: Check that Subnets have a VPC Endpoint for S3
+#
+# Trigger Type: Change Triggered
+# Scope of Changes: EC2:SubnetId
+# Accepted Parameters: None
+# Your Lambda function execution role will need to have a policy that provides the appropriate
+# permissions.  Here is a policy that you can consider.  You should validate this for your own
+# environment
+#{
+#    "Version": "2012-10-17",
+#    "Statement": [
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "logs:CreateLogGroup",
+#                "logs:CreateLogStream",
+#                "logs:PutLogEvents"
+#            ],
+#            "Resource": "arn:aws:logs:*:*:*"
+#        },
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "config:PutEvaluations",
+#                "ec2:DescribeVpcEndpoints"
+#            ],
+#            "Resource": "*"
+#        }
+#    ]
+#}
+#
+
+import boto3
+import botocore
+import json
+import logging
+
+
+APPLICABLE_RESOURCES = ["AWS::EC2::SubnetId"]
+
+log = logging.getLogger()
+log.setLevel(logging.INFO)
+
+def evaluate_compliance(configuration_item):
+        if configuration_item["resourceType"] not in APPLICABLE_RESOURCES:
+        return {
+            "compliance_type": "NOT_APPLICABLE",
+            "annotation": "The rule doesn't apply to resources of type " +
+            configuration_item["resourceType"] + "."
+               }
+    subnet_id   = configuration_item["configuration"]["subnetId"]
+    vpc_id      = configuration_item["configuration"]["vpcId"]
+    client      = boto3.client("ec2");
+    s3vpcendpt  = False
+    route_tables = []
+    for resource_type in configuration_item["relationships"]:
+        if resource_type['resourceType']=='AWS::EC2::RouteTable':
+            route_table = resource_type['resourceId']
+ 
+        
+    vpcendpoints = client.describe_vpc_endpoints()
+    
+    for m in vpcendpoints['VpcEndpoints']:
+        if m['ServiceName'].endswith('.s3') and m['State'] == 'available' and m['VpcId']== vpc_id:
+            route_tables= m['RouteTableIds']
```

Comment:
I don't know what the route table check is for. Can you explain how this factors into the rule? And maybe include it in the description if you feel like that is important